devito/function.py

```diff
@@ -256,7 +264,15 @@ def data_with_halo(self):
         The domain+halo data values.
 
         Elements are stored in row-major format.
+
+        .. note::
+
+            With this accessor you are claiming that you will modify
+            the values you get back. If you only need to look at the
+            values, use :meth:`data_ro_with_halo` instead.
         """
+        self._is_halo_dirty = True
```

Comment:
"dirty"?

Reply:
there are lots of flags attached to a Function, this one is about the halo, so I think we should keep "halo_dirty". 

Oh, maybe your question question was about the word "dirty" -- why "dirty". It's standard MPI halo exchange jargon: when you write to data, an update (through a halo exchange) is required so that the other processes have accesses to the most recent snapshot. Data that needs updating is commonly referred to as "dirty". Incidentally, this goes beyond MPI. It's for example used for cache lines too, to flag those that need to be committed back to DRAM.